tuna.go

```diff
@@ -360,32 +373,30 @@ func (c *Common) GetPrice() (common.Fixed64, common.Fixed64) {
 	return c.entryToExitPrice, c.exitToEntryPrice
 }
 
-func (c *Common) StartUDPReaderWriter(conn *net.UDPConn) {
+func (c *Common) StartUDPReaderWriter(conn udpconn.Conn) {
 	go func() {
 		for {
-			buffer := make([]byte, 2048)
-			n, err := conn.Read(buffer)
+			buffer := make([]byte, c.Service.UDPBufferSize)
+			n, _, err := conn.ReadFromUDP(buffer)
 			if err != nil {
 				log.Println("Couldn't receive data from server:", err)
 				if strings.Contains(err.Error(), "use of closed network connection") {
 					c.udpCloseChan <- struct{}{}
-					return
 				}
-				continue
+				return
 			}
 
-			data := make([]byte, n)
-			copy(data, buffer)
-			c.udpReadChan <- data
+			c.udpReadChan <- buffer[:n]
```

Comment:
The original code you removed copies the data before sending to channel, so there won't be any problem if buffer is changed (by reading new data) before channel data is consumed. This new code will have issue if buffer is changed before channel data is consumed.

Reply:
Code will block here until data consumed cause `c.udpReadChan` is a no buffer channel, then `buffer` can be changed during next `conn.ReadFromUDP(buffer)`

> The original code you removed copies the data before sending to channel, so there won't be any problem if buffer is changed (by reading new data) before channel data is consumed. This new code will have issue if buffer is changed before channel data is consumed.